macros/light-sdk-macros/tests/hasher.rs

```diff
@@ -15,64 +16,1202 @@ pub struct MyAccount {
-        f: None,
-    };
-    assert_eq!(
-        my_account.hash::<Poseidon>().unwrap(),
-        [
-            44, 62, 31, 169, 73, 125, 135, 126, 176, 7, 127, 96, 183, 224, 156, 140, 105, 77, 225,
-            230, 174, 196, 38, 92, 0, 44, 19, 25, 255, 109, 6, 168
-        ]
-    );
-
-    let my_account = MyAccount {
-        a: true,
-        b: u64::MAX,
-        c: MyNestedStruct {
-            a: i32::MIN,
-            b: u32::MAX,
-            c: "wao".to_string(),
-        },
-        d: [u8::MAX; 32],
-        e: MyNestedNonHashableStruct {
-            a: PhantomData,
-            b: Rc::new(RefCell::new(usize::MAX)),
-        },
-        f: Some(0),
-    };
-    assert_eq!(
-        my_account.hash::<Poseidon>().unwrap(),
-        [
-            32, 205, 141, 227, 236, 5, 28, 219, 24, 164, 215, 79, 151, 131, 162, 82, 224, 101, 171,
-            201, 4, 181, 26, 146, 6, 1, 95, 107, 239, 19, 233, 80
-        ]
-    );
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    /// Test suite for LightHasher derive macro
+    ///
+    /// Tests cover:
+    /// - Basic hashing: Byte representation, boundary values
+    /// - Attribute behavior:
+    ///   - #[truncate]: Arrays, strings, multiple fields
+    ///   - #[nested]: Recursive nesting, field counts
+    /// - Option handling:
+    ///   - Basic, truncated and nested variants
+    ///   - Uniqueness: None vs Some(0), different Some values, partial Some/None
+    ///   - Field ordering: Different field positions produce unique hashes
+    ///   - Truncated options: None vs empty, different values, varying lengths
+    /// - Error cases: Empty structs, Poseidon width limits
+    mod fixtures {
+        use super::*;
+
+        pub fn create_nested_struct() -> MyNestedStruct {
+            MyNestedStruct {
+                a: i32::MIN,
+                b: u32::MAX,
+                c: "wao".to_string(),
+            }
+        }
+
+        pub fn create_account(f: Option<usize>) -> MyAccount {
+            MyAccount {
+                a: true,
+                b: u64::MAX,
+                c: create_nested_struct(),
+                d: [u8::MAX; 32],
+                e: MyNestedNonHashableStruct {
+                    a: PhantomData,
+                    b: Rc::new(RefCell::new(usize::MAX)),
+                },
+                f,
+            }
+        }
+
+        pub fn create_zero_nested() -> MyNestedStruct {
+            MyNestedStruct {
+                a: 0,
+                b: 0,
+                c: "".to_string(),
+            }
+        }
+    }
+
+    mod basic_hashing {
+        use super::fixtures::*;
+        use super::*;
+
+        #[test]
+        fn test_byte_representation() {
+            let nested_struct = create_nested_struct();
+            let account = create_account(Some(42));
+
+            let manual_nested_bytes: Vec<Vec<u8>> = vec![
+                nested_struct.a.to_le_bytes().to_vec(),
+                nested_struct.b.to_le_bytes().to_vec(),
```

Comment:
the poseidon hasher hashes everything be not le we should use be in the macros as well.

Reply:
this will require changes across multiple crates. will change in a follow up pr + hold hasher release until then